src/lib.rs

```diff
@@ -270,8 +270,20 @@ fn open_browser_internal(browser: Browser, url: &str) -> Result<ExitStatus> {
     match browser {
         Browser::Default => open_on_unix_using_browser_env(url)
             .or_else(|_| -> Result<ExitStatus> { Command::new("xdg-open").arg(url).status() })
+            .or_else(|r| -> Result<ExitStatus> {
+                if let Ok(desktop) = ::std::env::var("XDG_CURRENT_DESKTOP") {
+                    if desktop == "KDE" {
+                        return Command::new("kioclient").arg("exec").arg(url).status();
+                    }
+                }
+                Err(r) // If either `if` check fails, fall through to the next or_else
+            })
             .or_else(|_| -> Result<ExitStatus> { Command::new("gvfs-open").arg(url).status() })
-            .or_else(|_| -> Result<ExitStatus> { Command::new("gnome-open").arg(url).status() }),
+            .or_else(|_| -> Result<ExitStatus> { Command::new("gnome-open").arg(url).status() })
+            .or_else(|_| -> Result<ExitStatus> {
+                Command::new("kioclient").arg("exec").arg(url).status()
+            })
+            .or_else(|_| -> Result<ExitStatus> { Command::new("x-www-browser").arg(url).status() }),
```

Comment:
[This page](https://bugzilla.redhat.com/show_bug.cgi?id=755553) seems to suggest that `x-www-browser` isn't usually available. Comments?

Reply:
Most Linux distros descent from two major lineages: Debian and Red Hat.

The alternatives system which provides `x-www-browser` is a feature of Debian-family distros (eg. Debian, Ubuntu, Mint, and all the lesser-known distros derived from them) and you linked to the Red Hat bug tracker.